### PR TITLE
Fixed `PYTHONPATH`'s `glob`

### PR DESCRIPTION
**Pull request by catap**
#### Description

This PR fixed a global issue with empty `PYTHONPATH` when it uses `glob`. So, this closes https://trac.macports.org/ticket/64045

Anyway, some affected ports cann't be tested localy:
 - `py27-ngl`: https://trac.macports.org/ticket/64046
 - `py27-numeric`: https://trac.macports.org/ticket/64047
 - `py39-gobject`: https://trac.macports.org/ticket/64048

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.6 20G165 x86_64
Xcode 13.1 13A1030d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
@ryandesign here it is

File: python/py-distorm/Portfile
Comment (reneeotten) [inline python/py-distorm/Portfile:35]: `test.env` needs to be in here, not `test.cmd`

Comment (catap) [reply python/py-distorm/Portfile]: @reneeotten thanks, I've forgot to make a force push. It should be fixed.

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
@stromnov @jmroot @petrrr @kurthindenburg 

Folks, given that these changes are all test-related, does anyone have any concerns with merging this?

We can certainly wait three days, for those interested in a closer look. But the gist is that all of these Python ports have the same change, which is the following:

Add a `pre-test` phase, setting `test.env` appropriately:

```
    pre-test {
        test.env        PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]
    }
```

Revert the prior commit, which set `test.env` unconditionally, even when not running tests:

```
test.env PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]
```

**Review by reneeotten — APPROVED** [review]

**Comment by kurthindenburg** [comment]
This is fine w/ me

**Comment by mascguy** [comment]
@catap Since today's a holiday for some folks - and it's only been two days - let's wait at least another full day or two before merging. (While I'd love to do it ASAP - particularly given that the changes are purely test-related - all of the maintainers should have a chance to give their blessing first.)

**Pull request merged**
